compiler/next/test/frontend/testParseStringBytesLiterals.cpp

```diff
@@ -0,0 +1,182 @@
+                             const std::string& expectValue) {
+  std::string toparse = "var x = b";
+  toparse += str;
+  toparse += ";\n";
+  auto parseResult = parser->parseString(testname, toparse.c_str());
+  assert(parseResult.errors.size() == 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModule());
+  auto module = parseResult.topLevelExpressions[0]->toModule();
+  assert(module->numStmts() == 1);
+  auto variable = module->stmt(0)->toVariable();
+  assert(variable);
+  auto bytesLit = variable->initExpression()->toBytesLiteral();
+  assert(bytesLit);
+  assert(bytesLit->quoteStyle() == expectQuoteStyle);
+  assert(bytesLit->str() == expectValue);
+}
+static void testCStringLiteral(Parser* parser,
+                               const char* testname,
+                               const char* str,
+                               StringLiteral::QuoteStyle expectQuoteStyle,
+                               const std::string& expectValue) {
+  std::string toparse = "var x = c";
+  toparse += str;
+  toparse += ";\n";
+  auto parseResult = parser->parseString(testname, toparse.c_str());
+  assert(parseResult.errors.size() == 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModule());
+  auto module = parseResult.topLevelExpressions[0]->toModule();
+  assert(module->numStmts() == 1);
+  auto variable = module->stmt(0)->toVariable();
+  assert(variable);
+  auto strLit = variable->initExpression()->toCStringLiteral();
+  assert(strLit);
+  assert(strLit->quoteStyle() == expectQuoteStyle);
+  assert(strLit->str() == expectValue);
+}
+
+static void testLiteral(Parser* parser,
+                        const char* testname,
+                        const char* str,
+                        StringLiteral::QuoteStyle expectQuoteStyle,
+                        const std::string& expectValue) {
+  testStringLiteral(parser, testname, str, expectQuoteStyle, expectValue);
+  testBytesLiteral(parser, testname, str, expectQuoteStyle, expectValue);
+}
+
+static void testBadLiteral(Parser* parser,
+                           const char* testname,
+                           const char* str) {
+  std::string toparse = "var x = ";
+  toparse += str;
+  toparse += ";\n";
+  auto parseResult = parser->parseString(testname, toparse.c_str());
+  assert(parseResult.errors.size() > 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModule());
+  auto module = parseResult.topLevelExpressions[0]->toModule();
+  assert(module->numStmts() == 1);
+  auto variable = module->stmt(0)->toVariable();
+  assert(variable);
+  assert(variable->initExpression()->isErroneousExpression());
+}
+
+int main() {
+  Context context;
+  Context* ctx = &context;
+
+  auto parser = Parser::build(ctx);
+  Parser* p = parser.get();
+
+  testLiteral(p, "test0.chpl", "\"hi\"",
```

Comment:
It might be nicer to read to instead have separate test functions that handle the single and triple quote forms themselves.

Reply:
So since you're doing the transformation into a full string above like this:
```C
std::string toparse = "var x = ";
toparse += str;
toparse += ";\n";
```
it seems like you could just as easily copy the test function and instead have one that writes:
```C
std::string toparse = "var x = \"";
toparse += str;
toparse += "\";\n";
```
and one that writes:
```C
std::string toparse = "var x = \"\"\"";
toparse += str;
toparse += "\"\"\";\n";
```
and that way you'd have less strings with `\"\"\"` in them over all because it would only happen in a test function rather than in every argument to a test function when you wanted to test triple quotes

Reply:
It'd also make it easier to see that you've covered everything, cuz in main you'd instead have:
```C
testOneSingleQuote(...);
testOneDoubleQuote(...);
testThreeSingleQuotes(...);
testThreeDoubleQuotes(...);
```
And it'd be easier to add coverage of a different inner string because you could just copy those four lines and replace with the same contents rather than having to individually adjust them